modules/statistics/php/charts.class.inc

```diff
@@ -111,20 +111,22 @@ class Charts extends \NDB_Page
         $user          = \NDB_Factory::singleton()->user();
         $list_of_sites = $user->getStudySites();
 
-        foreach ($list_of_sites as $siteID => $siteName) {
-            $totalRecruitment = $DB->pselectOne(
-                "SELECT COUNT(c.CandID)
+        $data = $DB->pselectColWithIndexKey(
+            "
+            SELECT COUNT(c.CandID), c.RegistrationCenterID as CenterID
                 FROM candidate c
-                WHERE
-                c.RegistrationCenterID=:Site AND
-                c.Active='Y' AND
-                c.Entity_type='Human'",
-                ['Site' => $siteID]
-            );
+                WHERE c.Active='Y' AND c.Entity_type='Human'
+                GROUP BY c.RegistrationCenterID
+        ",
+            [],
+            "CenterID"
+        );
+
+        foreach ($list_of_sites as $siteID => $siteName) {
 
             $recruitmentBySiteData[] = [
                 "label" => $siteName,
-                "total" => $totalRecruitment,
+                "total" => intval($data[$siteID]) ?? 0,
             ];
         }
```

Comment:
I think this could be further optimized by getting the site label in the query as well so that you don't have to have a loop through each site, and using COALESCE on the "COUNT(c.CandID)" with "0".

Reply:
I don't think this is a good idea because `$list_of_sites` is  `$user->getStudySites()`, not `\Utility::getSiteList()`, so doing this would require updating the query to get the user's study sites instead of trusting the logic that's already restricting it from PHP. This is fast enough to take <1s with the UK biobank, so I don't think it's necessary to take the risk of introducing bugs. Is it slow on CCNA?

Reply:
Makes sense! I don't feel strongly about the change so I think it's fine to leave